File: src/AS_23_12_week2/wangon/20291.py
n = int(input())
ret = {}

for i in range(n):
    instr = str(input())
    sep = instr.find('.')
    ext = instr[sep+1:len(instr)]
    if ext in ret:
        ret[ext] += 1
    else:
        ret[ext] = 1

for key, item in sorted(ret.items()):
    print(f"{key} {item}")

File: src/AS_23_12_week2/wangon/20291.cpp
#include <algorithm>
#include <cstdlib>
#include <iostream>
#include <map>
#include <sstream>
#include <string>
#include <vector>
using namespace std;

int main()
{
	int n;
	cin >> n;
	map<string, int> ret;
	for (int i = 0; i < n; i++)
	{
		string input, ext;
		size_t sep;
		cin >> input;
		sep = input.find('.');
		ext = input.substr(sep + 1, input.size());
		ret[ext] += 1;
	}
	for (auto const &pair : ret)
		cout << pair.first << " " << pair.second << endl;
}
